test/integration-test/src/tests/relocations.rs

```diff
@@ -0,0 +1,247 @@
+use anyhow::{Context, Result};
+use std::{process::Command, thread::sleep, time::Duration};
+
+use aya::{maps::Array, programs::TracePoint, BpfLoader, Btf, Endianness};
+
+use super::{integration_test, IntegrationTest};
+
+#[integration_test]
+fn relocate_field() {
+    let test = RelocationTest {
+        local_definition: r#"
+            struct foo {
+              __u8 a;
+              __u8 b;
+              __u8 c;
+              __u8 d;
+            };
+        "#,
+        target_btf: r#"
+            struct foo {
+              __u8 a;
+              __u8 c;
+              __u8 b;
+              __u8 d;
+            } s1;
+        "#,
+        relocation_code: r#"
+            __u8 memory[] = {1, 2, 3, 4};
+            __u32 value = BPF_CORE_READ((struct foo *)&memory, c);
+        "#,
+    }
+    .build()
+    .unwrap();
+    assert_eq!(test.run().unwrap(), 2);
+    assert_eq!(test.run_no_btf().unwrap(), 3);
+}
+
+#[integration_test]
+fn relocate_enum() {
+    let test = RelocationTest {
+        local_definition: r#"
+            enum foo { D = 1 };
+        "#,
+        target_btf: r#"
+            enum foo { D = 4 } e1;
+        "#,
+        relocation_code: r#"
+            __u32 value = bpf_core_enum_value(enum foo, D);
+        "#,
+    }
+    .build()
+    .unwrap();
+    assert_eq!(test.run().unwrap(), 4);
+    assert_eq!(test.run_no_btf().unwrap(), 0);
+}
+
+#[integration_test]
+fn relocate_pointer() {
+    let test = RelocationTest {
+        local_definition: r#"
+            struct foo {};
+            struct bar { struct foo *f; };
+        "#,
+        target_btf: r#""#,
+        relocation_code: r#"
+            __u8 memory[] = {0, 0, 0, 0, 0, 0, 0, 42};
+            struct foo *f = BPF_CORE_READ((struct bar *)&memory, f);
+            __u32 value = (f == (struct foo *) 42);
+        "#,
+    }
+    .build()
+    .unwrap();
+    assert_eq!(test.run().unwrap(), 1);
```

Comment:
this is missing a `run_no_btf()` run

Reply:
Fixed, except that I don't really know what we should test here. Basically we're just testing loading ends with no error.
Let me know if you have a better idea or if we can keep it as is.